listenbrainz/webserver/templates/user/scraper.js

```diff
@@ -147,9 +147,24 @@ function getLastFMPage(page) {
     xhr.onload = function () {
         if (/^2/.test(this.status)) {
             if (numberOfPages <= 1) {
-              numberOfPages = JSON.parse(this.response)['recenttracks']['@attr']['totalPages'];
+                var data = JSON.parse(this.response);
+                numberOfPages = data['recenttracks']['@attr']['totalPages'];
+
+                // initially set the stop page to the total number of pages
+                // we'll later update it if we're able to stop earlier because of previous imports
+                stopPage = numberOfPages;
+
+                if (data['recenttracks']['track'].length > 0) {
+                    // if date is present, use it
+                    if ('date' in data['recenttracks']['track'][0]) {
+                        maximumTimestampForImport = data['recenttracks']['track'][0]['date']['uts'];
+                    }
+                    else { // the latest listen is a playing_now, use current time
```

Comment:
I wonder if it is best to ignore playing_now in this context and simply not change the maximumTimestampForImport?